### DISCUSSION TITLE
opensky.history how to get data frequency for every 30 seconds rather than every second

### DISCUSSION THREAD

**Discussion by MICHELEVELLA** [xoolive/traffic#397] [Q&A] [answered]
Hi,

I'm trying to get data for London Heathrow and since there's a lot of data to work through it's taking a lot of time to download. I'm also interested in every 30 seconds for each flight. Currently, I'm doing the Lazy Iteration to filter the data every 30 seconds after downloading it raw. Can I get the data directly for every 30 seconds as it would make the download faster?

**Comment by xoolive** [comment, marked as answer]
Sorry for the late answer. There is no facility to do that the way you want at the moment. But the query can be written, see "Useful Query Examples" in https://opensky-network.org/impala-guide.

I don't have the bandwidth to implement it at the moment, but a pull request is welcome if you can. Note that the whole code for impala and trino is now part of https://github.com/open-aviation/pyopensky
